core/src/test/java/edu/ucr/cs/riple/core/ConfigurationTest.java

```diff
@@ -160,6 +175,61 @@ public void test_required_flags_for_downstream_dependency_analysis_cli() {
     assertEquals(downstreamInfo, config.downstreamInfo);
   }
 
+  @Test
+  public void test_config_files_have_different_uuid() {
+    Set<String> observed = new HashSet<>();
+    // Test for NullAway config
+    FixSerializationConfig config = new FixSerializationConfig();
+    Path nullawayConfigPath = testDir.resolve("nullaway.xml");
+    for (int i = 0; i < 5; i++) {
+      Utility.writeNullAwayConfigInXMLFormat(config, nullawayConfigPath.toString());
+      String uuid = getValueFromTag(nullawayConfigPath, "/serialization/uuid");
+      if (observed.contains(uuid)) {
+        throw new IllegalStateException(
+            "Duplicate UUID found for NullAway config: " + uuid + " in set: " + observed);
+      }
+      observed.add(uuid);
+    }
+    observed.clear();
+    // Test for Scanner config
+    Path scannerConfig = testDir.resolve("scanner.xml");
+    ModuleInfo moduleInfo = new ModuleInfo(testDir, nullawayConfigPath, scannerConfig);
+    for (int i = 0; i < 5; i++) {
+      Utility.setScannerCheckerActivation(moduleInfo, true);
+      String uuid = getValueFromTag(scannerConfig, "/scanner/uuid");
+      if (observed.contains(uuid)) {
+        throw new IllegalStateException(
+            "Duplicate UUID found for Scanner config: " + uuid + " in set: " + observed);
+      }
+      observed.add(uuid);
+    }
+  }
+
+  /**
+   * Helper method for reading value of a node located at /key_1/key_2/.../key_n (in the form of
+   * {@code Xpath} query) from a xml document at the given path.
+   *
+   * @param path Path to xml file.
+   * @param key Key to locate the value, can be nested in the form of {@code Xpath} query (e.g.
+   *     /key1/key2/.../key_n).
+   * @return The value in the specified keychain as {@code String}.
+   */
+  private static String getValueFromTag(Path path, String key) {
+    try {
+      DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+      DocumentBuilder builder = factory.newDocumentBuilder();
+      Document doc = builder.parse(Files.newInputStream(path));
+      doc.normalize();
+      XPath xPath = XPathFactory.newInstance().newXPath();
+      return ((Node) xPath.compile(key).evaluate(doc, XPathConstants.NODE)).getTextContent();
+    } catch (XPathExpressionException
+        | ParserConfigurationException
+        | IOException
+        | SAXException ex) {
+      throw new RuntimeException("Could not extract value from tag: " + key, ex);
```

Comment:
does adding a comma between key and ex imply they will be concatenated with a space between them?

Reply:
I am not sure if I fully understood your comment, `ex` will be here will be passed as the cause of the exception and is not part of the string.

Reply:
Oh lol. Forget it.

Reply:
No. This is attaching `ex` as the cause of the `RuntimeException`, see the docs [here](https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html#RuntimeException(java.lang.String,%20java.lang.Throwable)). This is a common pattern in Java and means that both the stack trace starting at this line, and that of the original exception (plus its error message) will be preserved.